netlist/src/design.rs

```diff
@@ -164,17 +169,19 @@ impl Design {
         self.metadata.borrow()
     }
 
-    pub(crate) fn add_metadata_string(&self, string: &str) -> MetaStringIndex {
-        self.metadata.borrow_mut().add_string(string)
+    pub fn add_metadata_string(&self, string: &str) -> MetaStringRef {
+        let index = self.metadata.borrow_mut().add_string(string);
+        self.metadata.borrow().ref_string(self, index)
     }
 
     pub(crate) fn ref_metadata_string(&self, index: MetaStringIndex) -> MetaStringRef {
         self.metadata.borrow().ref_string(self, index)
     }
 
-    pub(crate) fn add_metadata_item(&self, item: &MetaItem) -> MetaItemIndex {
+    pub fn add_metadata_item(&self, item: &MetaItem) -> MetaItemRef {
```

Comment:
Now returns an ItemRef, and those internal users who need indices can do `.index()`